website/archiver/tasks.py

```diff
@@ -39,40 +38,49 @@ def create_app_context():
 
 logger = get_task_logger(__name__)
 
+class ArchiverSizeExceeded(Exception):
+
+    def __init__(self, result, *args, **kwargs):
+        super(ArchiverSizeExceeded, self).__init__(*args, **kwargs)
+        self.result = result
+
+
+class ArchiverStateError(Exception):
+
+    def __init__(self, info, *args, **kwargs):
+        super(ArchiverStateError, self).__init__(*args, **kwargs)
+        self.info = info
+
 class ArchiverTask(celery.Task):
     abstract = True
     max_retries = 0
 
     def on_failure(self, exc, task_id, args, kwargs, einfo):
-        if not kwargs:
+        job = ArchiveJob.load(kwargs.get('job_pk'))
+        if job.status == ARCHIVER_FAILURE:
+            # already captured
             return
-        job = ArchiveJob.load(kwargs['job_pk'])
+        if not job:
```

Comment:
This check probably needs to go before the `if job.status == ARCHIVER_FAILURE` check in order to prevent an `AttributeError`.